docs: update contributing guidelines

vicbergquist opened the pull request:
<!-- Thank you for contributing to Sanity. Please read our [Code of Conduct](https://github.com/sanity-io/sanity/blob/next/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/sanity-io/sanity/blob/next/CONTRIBUTING.md) before submitting a PR.

To help us review your Pull Request, please make sure you follow the steps and guidelines below.

It's OK to open a Pull Request to start a discussion/ask for help, but it should then be created as a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/). 

Do not delete the instructional comments. -->

**Type of change (check at least one)**

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x]  Documentation (fix or update to documentation)
- [ ]  Maintenance
- [ ]  Other, please describe:

**Does this change require a documentation update? (Check one)**

- [ ]  Yes
- [x]  No

**Current behavior**

<!-- Reference/link the relevant issue and/or describe the behavior. -->
The current docs can be confusing for anyone trying to contribute to Sanity.

**Description**

<!-- Please include a summary of the changes this PR introduces and/or which issue is fixed. Please also include relevant motivation and context of why this PR is necessary. -->
Suggestions for changing the docs to make it easier for community contributors.

**Note for release**

<!-- Please include a high level summary of the changes this PR introduce. The intended audience is both editors and developers. If it's introducing a new feature, remember to link to docs/blogpost, if it's a bugfix, please describe the bug in non-technical terms (e.g. how a user/developer may have experienced it).
For inspiration, check out the release notes from an earlier release: https://github.com/sanity-io/sanity/releases/tag/v0.142.0 -->

**Checklist** 

- [x]  I have read the [Contributing Guidelines](https://github.com/sanity-io/sanity/blob/next/CONTRIBUTING.md)
- [x]  The PR title includes a link to the relevant issue
- [x]  The PR title is appropriately formatted: `[some-package] PR title (#123)`
- [x]  The code is linted
- [x]  The test suite is passing
- [x]  Corresponding changes to the documentation have been made

** Other comments**

Wrote this a while back, might be typos etc., but wanted a place to discuss these changes.

kmelve commented on CONTRIBUTING.md:
```diff
@@ -1,100 +1,178 @@
+
+## What is the Sanity Studio?
+
+The [Sanity Studio](https://www.sanity.io/docs/content-studio) is an open-source editing environment built with React.
```
```suggestion
The [Sanity Studio](https://www.sanity.io/docs/sanity-studio) is an open-source editing environment built with React.
```

kmelve commented on CONTRIBUTING.md:
```diff
@@ -1,100 +1,178 @@
+To submit a Feature Request, please make sure that a request like your own doesn't already exist by searching through our [open issues](https://github.com/sanity-io/sanity/issues), before you create a new one. If someone already submitted the same feature requests you wanted to submit, please let us know you also want this feature by reacting with/adding a :+1: on the original comment, or leave a ...
 
-Prefer squash + merge. If it makes sense to keep individual commits (e.g. different people has been working on the same feature), the pull request should be merged with the "Rebase + merge"-button. If possible, each individual commit message should be rewritten with the pull-request number in paranthesis, e.g. `[some-package] Fix stuff (#22)`
+If the same Feature Requests doesn't already exist, create a new issue and use the Feature Requests template. Be sure to provide all the necessary information about your new feature to help us understand what you are requesting.
```
```suggestion
If the same Feature Request doesn't already exist, create a new issue and use the Feature Requests template. Be sure to provide all the necessary information about your new feature to help us understand what you are requesting.
```

kmelve commented on CONTRIBUTING.md:
```diff
@@ -1,100 +1,178 @@
+# Setting up Sanity locally
 
-## Publishing a hotfix release from `master`
+Make sure you have **Node.js version 4 or newer** and **npm version 5 or newer**.
```
Aren't we on a higher minimal node version now?

vicbergquist replied:
Copy-pasted from the current docs

vicbergquist commented:
> Great! I added a few comments and suggestions. I'm not sure, but should we drop the definitive article in "the Sanity Studio", since “Sanity Studio" is a product name? As in, it's not "the Sanity content studio", but "Sanity Studio".

Thanks Knut! I've updated it according to your comments 😊 @saasen recently also looked at the guidelines so I've added him to review it as well.

saasen commented on CONTRIBUTING.md:
```diff
@@ -1,102 +1,181 @@
+# Setting up Sanity locally
 
-## Publishing a hotfix release from `master`
+Make sure you have _Node.js version 4 or newer_ and _npm version 5 or newer_.
```
I highly doubt this works in Node 4. Don't we require at least Node 8?

rexxars replied:
Should be node 10 now

saasen commented on CONTRIBUTING.md:
````diff
@@ -1,102 +1,181 @@
 
-Any security issues should be submitted directly to security@sanity.io. In order to determine whether you are dealing with a security issue, ask yourself these two questions:
+```
+npm run start
````
We should probably encourage contributors to use `npm run watch`?

saasen commented on CONTRIBUTING.md:
```diff
@@ -1,102 +1,181 @@
+## Working on fixes or features
 
-If the answer to either of those two questions are "yes", then you're probably dealing with a security issue. Note that even if you answer "no" to both questions, you may still be dealing with a security issue, so if you're unsure, just email us at security@sanity.io.
+Please create a new branch branch for the fix or feature/enhancement you want to work on and regularly push your work to this branch. Remember to also keep your branch up to date by rebasing it regularly against the original `next` branch. To do this, add the original Sanity repository as a remote (in our example we called it `upstream`).
```
Typo: `branch branch`.

saasen commented on CONTRIBUTING.md:
```diff
@@ -1,102 +1,181 @@
 
-- Can I access something that's not mine, or something I shouldn't have access to?
-- Can I disable something for other people?
+## Working on fixes or features
```
Would it be wise to mention what kind of naming we want on the branch? We could for example say that they must start with `contribution/<branch-name-here>` or something similar.

saasen commented on CONTRIBUTING.md:
```diff
@@ -1,102 +1,181 @@
-- What did you do?
-- What did you expect to see?
-- What did you see instead?
+Please note that if you would like to develop a new feature for the Sanity Studio, it's unlikely that we will be able to accept your contribution. However, we highly recommend that you develop and publish your new feature as a plugin. Plugins can be [parts](https://www.sanity.io/docs/extending/parts), [custom input components](https://www.sanity.io/docs/extending/custom-input-widgets), [dashboard ...
```
GitHub shows this paragraph really strange. I don't know if it was something we did.

saasen replied:
Should we also mention that if they throw us a message beforehand and discuss their "new feature" or similar in an issue, we might accept it? Discussions started from the community might be very valuable input for stuff we create anyways, and if we encourage them to start a discussion in an issue, they would also feel more welcomed.

saasen commented on CONTRIBUTING.md:
```diff
@@ -1,102 +1,181 @@
+- To work on a fix or something new, create a descriptively named branch off of `next` (ie: feature/new-oauth2-scopes)
+- Commit to that branch locally and regularly push your work to the same named branch
+- Rebase your feature branch regularly against `next`. Make sure its even with `next` before creating a Pull Request or merging
+- Pull Requests should be as ready as possible for merge and the Pull Request template should be filled out to the best of your ability or in a way that is appropriate. Unless stated otherwise, it should also be safe to assume that:
```
I this this is somewhat wrong. It's possible to submit a draft PR. This to receive feedback on the ongoing development of a fix or similar. I think we should encourage people to ask us for guidance when they need it. There might be a lot of questions regarding the code base when working on a fix. Like Espen said, it's not good that they spend a lot of time and submitting a PR at the end, and if it isn't solved the way we want, we need to decline it. I think it's highly unlikely that they would submit another one after that.

saasen commented:
If we want to accept PRs in the CLI, we definitely need to write something around how to properly run it against a Studio. It's not straight forward at all.

saasen commented on CONTRIBUTING.md:
```diff
@@ -1,100 +1,239 @@
+- New features
+- Documentation improvements
+
+New features are unlikely to be accepted and are better developed and published as plugins that extend the existing functionality of the Sanity Studio. See [New features](#new-features) for more info.
```
Intentionally repeated as of L21?

saasen commented on CONTRIBUTING.md:
```diff
@@ -1,100 +1,239 @@
+
+### Bug reports
+
+To submit a bug report, check if the issue has already been reported, and if not, please create a new issue and use the provided template. Make sure your report answers these five questions:
```
Should we just point to the template and let that tell what needs to be answered?

saasen commented on CONTRIBUTING.md:
```diff
@@ -1,100 +1,239 @@
+
+If the same Feature Request doesn't already exist, create a new issue and use the Feature Requests template. Be sure to provide all the necessary information about your new feature to help us understand what you are requesting.
+
+# Setting up Sanity locally
```
Should we say something about the `symlinkDependencies` script in order to use local Sanity packages in an existing studio?

saasen commented on CONTRIBUTING.md:
```diff
@@ -1,100 +1,239 @@
+
+Please note that if you would like to develop a new feature for the Sanity Studio, it's unlikely that we will be able to accept your contribution. However, we highly recommend that you develop and publish your new feature as a plugin. Plugins can be [parts](https://www.sanity.io/docs/extending/parts), [custom input components](https://www.sanity.io/docs/extending/custom-input-widgets), [dashboard ...
+
+### Pull Requests
```
Should we say something about adding tests should be semi-required? Tests aren't setup everywhere yet, so maybe it would be a little harsh to require it. At the same time, adding them would be highly appreciated. 👍

saasen reviewed (approved):
LGTM! Added some comments as well.
